### PR TITLE
fix(rti): clarification of usage of federationId and simulationId

### PR DESCRIPTION
**Pull request by fhilg**
## Type of this PR 

- [ ] Bug fix
- [x] Enhancement

## Description

- rename: `MosaicSimulation.scenarioId` -> `MosaicSimulation.federationId`
- removal of unnecessary local variable `simId` in `MosaicSimulation.createFederation()`

## Issue(s) related to this PR

 * internal issue 384
 
 
## Affected parts of the online documentation

Are there any parts in the online documentation which are affected by this PR?

## Definition of Done

- [x] You have read CONTRIBUTING.md carefully.
- [x] You have signed the [Contributor License Agreement](http://www.eclipse.org/legal/CLA.php).
- [ ] All commits are signed-off (`-s`) with your mail address of your Eclipse Account.
- [x] `origin/main` has been merged into your Fork.
- [x] New functionality is covered by unit tests or integration tests. Code coverage must not decrease.
- [x] There are no new SpotBugs warnings. 
- [x] Coding guidelines have been observed (see CONTRIBUTING.md).
- [x] All tests pass.

### CODE REVIEW

File: rti/mosaic-starter/src/main/java/org/eclipse/mosaic/starter/MosaicSimulation.java [hunk 1]
```diff
@@ -169,12 +169,12 @@ public SimulationResult runSimulation(Path scenarioDirectory, CScenario scenario
 
             Validate.isTrue(Files.exists(scenarioDirectory), "Scenario directory at '" + scenarioDirectory + "' does not exist.");
 
-            scenarioId = Validate.notBlank(scenarioConfiguration.simulation.id,
+            federationId = Validate.notBlank(scenarioConfiguration.simulation.id,
                     "No simulation id given in scenario configuration file"
```
Comment (kschrab) [inline rti/mosaic-starter/src/main/java/org/eclipse/mosaic/starter/MosaicSimulation.java:173]: Rename error message to "No scenario id given ..."

File: rti/mosaic-starter/src/main/java/org/eclipse/mosaic/starter/MosaicSimulation.java [hunk 2]
```diff
@@ -442,13 +440,13 @@ private ComponentProvider createFederation(final MosaicComponentParameters simul
         TimeManagement time = componentProvider.getTimeManagement();
 
         if (watchdogInterval > 0) {
-            final WatchDog watchDogThread = time.startWatchDog(simId, watchdogInterval);
+            final WatchDog watchDogThread = time.startWatchDog(federationId, watchdogInterval);
```
Comment (kschrab) [inline rti/mosaic-starter/src/main/java/org/eclipse/mosaic/starter/MosaicSimulation.java:443]: I think here we should pass `simulationId` instead.

File: rti/mosaic-starter/src/main/java/org/eclipse/mosaic/starter/MosaicSimulation.java (hunk 1 above)
Comment (fhilg) [reply rti/mosaic-starter/src/main/java/org/eclipse/mosaic/starter/MosaicSimulation.java:173]: Keep in mind, the `scenario_config.json` looks like:
```json
{
    "simulation": {
        "id": "Tiergarten",
        "duration": "60s",
        ...
```

Comment (kschrab) [reply rti/mosaic-starter/src/main/java/org/eclipse/mosaic/starter/MosaicSimulation.java:173]: I know :( ok, then keep it as it is

**Review by kschrab — APPROVED** [review]

**Pull request merged**